docs/api/domains.md

```diff
@@ -23,6 +24,67 @@ Domains do not enforce any particular structure. However specific
 methods can be defined on domains to configure behavior at key points
 in a Microcosm's lifecycle.
 
+## Creating Domains
+
+There are two ways to create a domains: as a class, and as a plain object. The
+usage is roughly the same for both versions, the class form can additionally
+take advantage of having a `constructor`.
+
+### Domains as classes
+
+```javascript
+class Domain {
+  setup (repo, options) {
+    // Run startup behavior
+  }
+  teardown (repo) {
+    // Clean up any setup behavior
+  }
+  handleAction (state, payload) {
+    // Old state in, new state out...
+    let newState = { ...state, prop: payload.prop }
+
+    return newState
+  }
+  register () {
+    return {
+      [action] : this.handleAction
+    }
+  }
+}
+
+repo.addDomain('key', Domain)
+```
+
+### Domains as plain objects
+
+```javascript
+const Domain = {
+  setup (repo, options) {
+    // Run startup behavior
+  },
+  teardown (repo) {
+    // Clean up any setup behavior
+  },
+  handleAction (state, payload) {
+    let newState = { ...state, prop: payload.prop }
+
+    return newState
+  },
+  register () {
+    return {
+      [action] : this.handleAction
+    }
```

Comment:
There should be a mention that the registering can happen like this or how it is shown later (splitting it into it's lifecycle keys). Since this is already a confusing part seeing it written differently all over might be confusing.